model/channel.go

```diff
@@ -84,6 +84,21 @@ type DirectChannelForExport struct {
 	Members *[]string
 }
 
+type ChannelModeration struct {
+	Name  string                 `json:"name"`
+	Roles *ChannelModeratedRoles `json:"roles"`
+}
+
+type ChannelModeratedRoles struct {
+	Guests  *ChannelModeratedRole `json:"guests"`
+	Members *ChannelModeratedRole `json:"members"`
+}
+
+type ChannelModeratedRole struct {
+	Value   bool `json:"value"`
+	Enabled bool `json:"enabled"`
+}
+
```

Comment:
Decided to break these up into multiple structs instead of having one struct with maps